src/tools/runk/src/commands/exec.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2021-2022 Kata Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use anyhow::anyhow;
+use anyhow::Result;
+use libcontainer::status::Status;
+use libcontainer::{
+    builder::ContainerBuilder, container::ContainerAction, status::get_current_container_state,
+};
+use liboci_cli::Exec;
+use oci::{ContainerState, Process as OCIProcess, Spec};
+use rustjail::container::update_namespaces;
+use slog::{debug, info, Logger};
+use std::fs::File;
+use std::path::{Path, PathBuf};
+
+pub async fn run(opts: Exec, root: &Path, logger: &Logger) -> Result<()> {
+    let status = Status::load(root, &opts.container_id)?;
+    let state = get_current_container_state(&status)?;
+
+    // if state is Created or Running, we can execute the process. Otherwise, return an error.
+    match state {
+        ContainerState::Created | ContainerState::Running => (),
+        _ => return Err(anyhow!("cannot exec in a stopped or paused container")),
+    }
+
+    let ctx = ContainerBuilder::default()
+        .id(opts.container_id.clone())
+        .bundle(status.bundle)
+        .root(root.to_path_buf())
+        .console_socket(opts.console_socket.clone())
+        .pid_file(opts.pid_file.clone())
+        .spec(Some(adapt_exec_spec(
+            status
+                .config
+                .spec
+                .ok_or_else(|| anyhow!("cannot load spec from a running container"))?,
+            status.pid,
+            opts,
+            logger,
+        )?))
+        .build()?
+        .create_ctx()?;
+    info!(&logger, "create ctx successfully");
```

Comment:
Should we keep this log?

Reply:
I use it to debug .I will remove it and add some more specific debug output at some right place.